npm/vite-dev-server/src/devServer.ts

```diff
@@ -24,10 +23,9 @@ export async function devServer (config: ViteDevServerConfig): Promise<Cypress.R
   const server = await devServer.create(config, vite)
 
   debug('Vite server created')
-  const port = await getPort({ port: 3000 })
 
-  debug('Starting Vite Server on port %i', port)
-  await server.listen(port)
+  await server.listen()
+  const port = server.config.server.port!
```

Comment:
Kinda minor, but the Vite docs seem ambiguous about if port exists when you haven't specified a port to `server.listen()`:

```js
    /**
     * Specify server port. Note if the port is already being used, Vite will
     * automatically try the next available port so this may not be the actual
     * port the server ends up listening on.
     */
    port?: number;
```

So maybe it would be a good idea to guard against `undefined`, just in case the implementation changes and this is no longer available immediately after `await server.listen()`.

```suggestion
  const port = server.config.server.port!

  if (!port) throw new Error('Missing vite dev server port.')
```

Reply:
I like this, I think we generally error loud and clear rather than assuming something that's absolutely required will always exist. Better debugging and bug reports!

Reply:
Done in [f4d92cb](https://github.com/cypress-io/cypress/pull/21490/commits/f4d92cbc1b723842a71176717fbbff1bc253aa2a)!